rtt_rosparam/CMakeLists.txt

```diff
@@ -12,14 +12,24 @@ include_directories(
   ${catkin_INCLUDE_DIRS}
   ${EIGEN3_INCLUDE_DIRS})
 
-# build ROS param as a separate service
+# Add the data source header
+orocos_library(ros_param_data_source src/ros_param_data_source.cpp)
+target_link_libraries(ros_param_data_source
+  ${catkin_LIBRARIES}
+)
+
+  # build ROS param as a separate service
 orocos_service(rtt_rosparam src/rtt_rosparam_service.cpp)
 target_link_libraries(rtt_rosparam 
 #  ${OROCOS-RTT_RTT-MARSHALLING_LIBRARY} 
-  ${catkin_LIBRARIES})
+  ${catkin_LIBRARIES}
+  ros_param_data_source
+)
```

Comment:
With the unnecessary source file removed, this change must be reverted. As a template class, `RosParamDataSource<T>` is header-only.